### PR TITLE
Feature/condo/sberdoma 58/update ticket create update

### PR DESCRIPTION
**Pull request by Dimitreee**

### CODE REVIEW

File: apps/condo/components/BaseTicketForm/index.tsx [hunk 1]
```diff
@@ -232,8 +141,13 @@ export const BaseTicketForm:React.FC<ITicketFormProps> = (props) => {
...
                                                     <Col span={11}>
-                                                        <Form.Item name={'clientPhone'} rules={validations.clientPhone} label={PhoneLabel} validateFirst>
-                                                            <MaskedInput mask={'+1 (111) 111-11-11'} placeholderChar={'0'}/>
+                                                        <Form.Item
+                                                            name={'clientPhone'}
+                                                            rules={validations.clientPhone}
+                                                            label={PhoneLabel}
+                                                            validateFirst
+                                                        >
+                                                            <NumericInput/>
```
Comment (pahaz) [inline apps/condo/components/BaseTicketForm/index.tsx:150]: PhoneInput?

File: apps/condo/components/FormResetButton.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    const intl = useIntl()
+
+    return (
+        <Form.Item noStyle shouldUpdate>
+            {({ isFieldsTouched, resetFields }) => {
+                const isTouched = isFieldsTouched()
+
+                return (
+                    <Button {...props} onClick={() => resetFields()} disabled={!isTouched}>
+                        {intl.formatMessage({ id: 'Cancel' })}
```
Comment (pahaz) [inline apps/condo/components/FormResetButton.tsx:16]: new translation style

File: apps/condo/components/LinkWithIcon.tsx [hunk 3]
```diff
@@ -45,12 +49,10 @@ export const LinkWithIcon = (props: ILinkWithIconProps) => {
...
         <StyledButton type='link' onClick={handleClick}>
-            <Space size={16}>
-                {props.icon && <IconContainer className='icon'>{props.icon}</IconContainer>}
-                <Typography.Text className='text' style={{ fontSize: '12px' }}>
-                    {intl.formatMessage({ id: props.locale })}
-                </Typography.Text>
-            </Space>
+            {props.icon && <IconContainer className='icon'>{props.icon}</IconContainer>}
+            <Typography.Text className='text'>
+                {intl.formatMessage({ id: props.locale })}
```
Comment (pahaz) [inline apps/condo/components/LinkWithIcon.tsx:54]: is it better to use `children` here?

File: apps/condo/components/NumericInput.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { Input, InputProps } from 'antd'
+import React from 'react'
+import { NUMBERS_AND_PLUS } from '../constants/regexps'
+
+export const NumericInput:React.FC<InputProps> = (props) => {
```
Comment (pahaz) [inline apps/condo/components/NumericInput.tsx:5]: PhoneNumberInput

File: apps/condo/components/LinkWithIcon.tsx [hunk 5]
```diff
@@ -45,12 +49,10 @@ export const LinkWithIcon = (props: ILinkWithIconProps) => {
 
     return (
         <StyledButton type='link' onClick={handleClick}>
-            <Space size={16}>
-                {props.icon && <IconContainer className='icon'>{props.icon}</IconContainer>}
-                <Typography.Text className='text' style={{ fontSize: '12px' }}>
-                    {intl.formatMessage({ id: props.locale })}
-                </Typography.Text>
-            </Space>
+            {props.icon && <IconContainer className='icon'>{props.icon}</IconContainer>}
```
Comment (pahaz) [inline apps/condo/components/LinkWithIcon.tsx:52]: You need to remove Space from imports. Antd style assumes to use the Space component here

File: apps/condo/constants/regexps.js [hunk 6]
```diff
@@ -1,7 +1,9 @@
 const UPPER_CASE_ALPHANUMERIC_REGEXP = /^[A-Z0-9_]+$/
 const LETTERS_AND_NUMBERS = /[\p{L}\p{N}]/gu
+const NUMBERS_AND_PLUS = /^\+?\d*(\.\d*)?$/
```
Comment (pahaz) [inline apps/condo/constants/regexps.js:3]: Phone?

File: apps/condo/containers/TicketForm/CreateTicketForm.tsx [hunk 7]
```diff
@@ -1,23 +1,80 @@
...
 
 export const CreateTicketForm:React.FC = () => {
+    const intl = useIntl()
+    const CreateTicketMessage = intl.formatMessage({ id: 'CreateTicket' })
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
     const { organization } = useOrganization()
     const router = useRouter()
+    // TODO(Dimitreee):remove after typo inject
+    const auth = useAuth() as { user: {id:string} }
```
Comment (pahaz) [inline apps/condo/containers/TicketForm/CreateTicketForm.tsx:27]:
```suggestion
    const auth = useAuth() as { user: { id:string } }
```

File: apps/condo/containers/TicketForm/CreateTicketForm.tsx [hunk 8]
```diff
@@ -1,23 +1,80 @@
...
 
-    return (<BaseTicketForm action={action} initialValues={{}} organization={organization}/>)
+                                return (
+                                    <Row gutter={[0, 24]}>
+                                        <ErrorsContainer property={property} unitName={unitName}/>
+                                        <Col span={24}>
+                                            <Button
+                                                key='submit'
+                                                onClick={handleSave}
+                                                type='sberPrimary'
```
Comment (pahaz) [inline apps/condo/containers/TicketForm/CreateTicketForm.tsx:64]: just primary?

File: apps/condo/containers/TicketForm/UpdateTicketForm.tsx [hunk 9]
```diff
@@ -36,6 +45,41 @@ export const UpdateTicketForm:React.FC<IUpdateTicketForm> = ({ id }) => {
             action={updateAction}
             initialValues={Ticket.convertGQLItemToFormState(obj)}
             organization={organization}
-        />
+        >
+            {({ handleSave, isLoading }) => {
```
Comment (pahaz) [inline apps/condo/containers/TicketForm/UpdateTicketForm.tsx:49]: Why we have two forms for create/update?

File: apps/condo/lang/ru.json [hunk 10]
```diff
@@ -28,10 +28,10 @@
   "pages.condo.ticket.field.Type": "Тип заявки",
   "pages.condo.ticket.field.Details": "Детали проблемы",
   "pages.condo.ticket.field.Status": "Статус",
-  "placeholder.Address": "Улица, дом",
-  "placeholder.Description": "Опишите суть проблемы как можно подробнее",
   "field.Source": "Источник",
   "field.Address": "Адрес",
+  "placeholder.Address": "Улица, дом",
```
Comment (pahaz) [inline apps/condo/lang/ru.json:33]: !! reorder in one file !!

File: apps/condo/utils/ticket.ts [hunk 11]
```diff
@@ -80,4 +80,22 @@ export const getTicketLabel = (intl, ticket) => {
     )
 
     return `${get(ticket, ['status', 'name'])} ${intl.formatMessage({ id: 'From' })} ${formattedDate}`
+}
+
+export const sortStatusesByType = (statuses) => {
+    // status priority map [min -> max]
+    const orderedStatusPriority = ['deferred', 'deferred', 'canceled', 'completed', 'processing', 'new_or_reopened' ]
```
Comment (Dimitreee) [inline apps/condo/utils/ticket.ts:87]: ....fix it

File: apps/condo/pages/ticket/[id]/update.tsx [hunk 12]
```diff
@@ -40,12 +39,19 @@ const TicketUpdatePage = () => {
...
-TicketUpdatePage.headerAction = (
-    <LinkWithIcon
-        icon={<ArrowLeftOutlined style={{ color: colors.white }}/>}
-        locale={'menu.AllTickets'}
-        path={'/ticket/'}
-    />
-)
+const HeaderAction = () => {
+    const { query } = useRouter()
+    console.log(query)
```
Comment (pahaz) [inline apps/condo/pages/ticket/[id]/update.tsx:44]: debug detected

File: apps/condo/pages/ticket/[id]/update.tsx [hunk 13]
```diff
@@ -40,12 +39,19 @@ const TicketUpdatePage = () => {
...
+
+    return (
+        <LinkWithIcon
+            icon={<ArrowLeftOutlined style={{ color: colors.white }}/>}
+            locale={'Back'}
```
Comment (pahaz) [inline apps/condo/pages/ticket/[id]/update.tsx:49]: It is not clear that this text will be used in translations. Abstraction leak

Comment (pahaz) [reply apps/condo/pages/ticket/[id]/update.tsx:49]: It's better to use similar to AntD interface to the reusable atom components

File: apps/condo/containers/TicketForm/UpdateTicketForm.tsx (hunk 9 above)
Comment (Dimitreee) [reply apps/condo/containers/TicketForm/UpdateTicketForm.tsx:49]: there may be different logic before / after creation / editing of ticket

**Pull request merged**